packages/postgres-database/src/simcore_postgres_database/utils_groups_extra_properties.py

```diff
@@ -0,0 +1,150 @@
+from .utils_models import FromRowMixin
+
+
+class GroupExtraPropertiesError(Exception):
+    ...
+
+
+class GroupExtraPropertiesNotFound(GroupExtraPropertiesError):
+    ...
+
+
+@dataclass(frozen=True, slots=True, kw_only=True)
+class GroupExtraProperties(FromRowMixin):
+    group_id: int
+    product_name: str
+    internet_access: bool
+    override_services_specifications: bool
+    created: datetime.datetime
+    modified: datetime.datetime
+
+
+async def _list_table_entries_ordered_by_group_type(
+    connection: SAConnection, user_id: int, product_name: str
+) -> list[RowProxy]:
+    subquery = (
+        sqlalchemy.select(
+            *[
+                groups_extra_properties,
+                groups.c.type,
+                sqlalchemy.case(
+                    *[
+                        # NOTE: the ordering is important for the aggregation afterwards
+                        (groups.c.type == "EVERYONE", sqlalchemy.literal(3)),
+                        (groups.c.type == "STANDARD", sqlalchemy.literal(2)),
+                        (groups.c.type == "PRIMARY", sqlalchemy.literal(1)),
+                    ],
+                    else_=sqlalchemy.literal(4),
+                ).label("type_order"),
+            ]
+        )
+        .select_from(
+            sqlalchemy.join(
+                sqlalchemy.join(
+                    groups_extra_properties,
+                    user_to_groups,
+                    groups_extra_properties.c.group_id == user_to_groups.c.gid,
+                ),
+                groups,
+                groups_extra_properties.c.group_id == groups.c.gid,
+            )
+        )
+        .where(
+            (groups_extra_properties.c.product_name == product_name)
+            & (user_to_groups.c.uid == user_id)
+        )
+        .alias()
+    )
+
+    result = await connection.execute(
+        sqlalchemy.select(subquery).order_by(subquery.c.type_order)
+    )
+    assert result  # nosec
+
+    rows: list[RowProxy] | None = await result.fetchall()
+    assert rows is not None  # nosec
+    return rows
+
+
+def _merge_extra_properties_booleans(
+    instance1: GroupExtraProperties, instance2: GroupExtraProperties
+) -> GroupExtraProperties:
+    merged_properties: dict[str, Any] = {}
+    for field in fields(instance1):
+        value1 = getattr(instance1, field.name)
+        value2 = getattr(instance2, field.name)
+
+        if isinstance(value1, bool):
+            merged_properties[field.name] = value1 or value2
+        else:
+            merged_properties[field.name] = value1
+    return GroupExtraProperties(**merged_properties)  # pylint: disable=missing-kwoa
+
+
+@dataclass(frozen=True, slots=True, kw_only=True)
+class GroupExtraPropertiesRepo:
+    @staticmethod
+    async def get(
+        connection: SAConnection, *, gid: int, product_name: str
+    ) -> GroupExtraProperties:
+        get_stmt = sqlalchemy.select(groups_extra_properties).where(
```

Comment:
Q: why you use full `sqlachemy` vs shortcut `sa`? is there any reason we should prefer the former?

Reply:
this was due to this issues with mypy/pylance. but here I guess there is no reason. I have something big coming regarding this.